ocaml/lambda/lambda.ml

```diff
@@ -539,6 +553,41 @@ let make_key e =
     Some (tr_rec Ident.empty e)
   with Not_simple -> None
 
+let join_actions_to_be_shared ~printer:_ lam1 lam2 =
+  (* [printer] is left here for future debugging purposes. *)
+  match lam1, lam2 with
+  (* Since the keys for these actions must have compared equal then the
+     structure of [lam1] and [lam2] must be the same. *)
+  | Llet (let_kind, kind, bound, defining_expr,
+      Lprim (Pfield (index, sem1), args, scoped_loc1)),
+    Llet (_let_kind, _kind, _bound, _defining_expr,
+      Lprim (Pfield (_, sem2), _, scoped_loc2)) ->
+    (* This is specifically to catch cases like:
+        type t =
+          | A of { x : int; }
+          | B of { mutable x : int; }
+          | C of { x : int; }
+
+        let get t =
+          match t with
+          | A r -> r.x
+          | B r -> r.x
+          | C r -> r.x
+    *)
+    let sem = join_field_read_semantics sem1 sem2 in
+    let scoped_loc =
```

Comment:
Can't we just always take the debuginfo -- and indeed everything except the read semantics -- from the first one?